src/structstore_field.hpp

```diff
@@ -152,7 +152,13 @@ class StructStoreField {
         type = FieldType<T>::value;
     }
 
-    StructStoreField& operator=(StructStoreField&&) = delete;
+    StructStoreField& operator=(StructStoreField&& other) {
+        data = other.data;
+        type = other.type;
+        other.data = nullptr;
+        other.type = FieldTypeValue::EMPTY;
```

Comment:
```suggestion
        std::swap(data, other.data);
        std::swap(type, other.type);
```
Otherwise the memory in `this->data` might be leaked (deallocation is not possible as we don't have a pointer to the allocator here)